### PR TITLE
Bring histograms back to the future

### PR DESCRIPTION
**Pull request by wchargin**
Summary:
This commit removes the histogram dashboard's dependencies on old-style
framework code, and removes mentions of histograms from core code like
`backend.ts`.

As usual (e.g., #108), I had to delete some minimal tests in
`backendTests.ts` to do this. Unfortunately, this time there were also
some substantive tests of the `convertBins` function, which I regret
leaving behind. I've included a TODO in the code to add them back in
once our frontend tests work again, which will of course be Soon™.

Note that both the old and new versions of the histogram dashboard have
an unfortunate behavior regarding graphical updates after changing run
selection state. Specifically, suppose that Histogram A and Histogram B
appear side-by-side, and the run for Histogram A is toggled off. Then we
would like Histogram B to seamlessly move into the position of
Histogram A. However, due to Polymer's data binding, the actual action
is that Histogram B is deleted and Histogram A's data bindings are set
to those of the former Histogram B. The result is that the title, tag,
and card color of Histogram A immediately update to those of
Histogram B, but the actual data takes a while to fetch. Furthermore,
this means that _all_ histograms that are affected will re-render their
data. (So if the line goes down to Histogram Z, then removing the run
for Histogram A will cause 25 network requests.)

Essentially, we want [React's keys behavior][1], but it doesn't look
like we can get it. I'm leaving this wontfix for now.

[1]: https://facebook.github.io/react/docs/lists-and-keys.html#keys

Finally, while I'm at it, I fixed all closure compiler warnings in
`vz-histogram-timeseries` and `tf-histogram-dashboard`.

Test Plan:
Tested with the MNIST data combined with the histogram demo data.
Toggled runs, changed histogram modes and time modes, used tag groups,
expanded and collapsed categories. All bazel tests pass, but there
aren't actually any of interest, so that really just means that
everything still builds.

wchargin-branch: refactor-histograms-dashboard

### CODE REVIEW

File: tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html [hunk 1]
```diff
@@ -114,50 +119,71 @@
...
+        this.reload();
+      },
+      reload() {
+        this._fetchTags().then(() => {
+          this._reloadHistograms();
+        });
+      },
+      _fetchTags() {
+        const url = getRouter().pluginRoute('histograms', '/tags');
+        return this._requestManager.request(url).then(runToTag => {
```
Comment (dandelionmane) [inline tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html]: Is there a reason we don't use cancellable here?

File: tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-loader.html [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+       * (b) we weren't running TypeScript tests anyway and. Once we get
+       * TypeScript testing working again, reinstate these tests in a
+       * more appropriate location.
+       *
+       * @return A list of histogram bins. Each bin has an `x` (left
+       *     edge), a `dx` (width), and a `y` (count). If the given
+       *     right edges are inclusive, then these left edges (`x`) are
+       *     exclusive.
+       */
+      _convertBins(histogram, min, max, numBins) {
```
Comment (dandelionmane) [inline tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-loader.html]: Can we:
- Keep this file in Typescript - no need to lose the typing info

- Port the relevant tests to something like histogramTests.ts - even if it doesn't run yet, it's less likely that we'll forget to activate those tests if we have the file lying around, vs. remembering to look at the TODO

File: tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html (hunk 1 above)
Comment (dandelionmane) [inline tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html]: Either way

Comment (wchargin) [inline tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html]: Okay; will do them all at once as a New Feature, then.

Comment (dandelionmane) [inline tensorboard/plugins/histograms/tf_histogram_dashboard/tf-histogram-dashboard.html]: 👍

**Pull request merged**